### PR TITLE
chore: remove promisification deprecation callbacks

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Remove promisification deprecations and tests verifying callback versions. I think i got everything but please let me know if i missed any of them!

cc @miniak @MarshallOfSound 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Removed deprecations for asynchronous functions to force use of promises.

### CODE REVIEW

**Comment by miniak** [comment]
this is missing:
```
diff --git a/lib/browser/api/app.ts b/lib/browser/api/app.ts
index 0d6110e4c..625581e1b 100644
--- a/lib/browser/api/app.ts
+++ b/lib/browser/api/app.ts
@@ -89,9 +89,6 @@ for (const name of events) {
   })
 }
 
-// Function Deprecations
-app.getFileIcon = deprecate.promisify(app.getFileIcon)
-
 // Property Deprecations
 deprecate.fnToProperty(app, 'accessibilitySupportEnabled', '_isAccessibilitySupportEnabled', '_setAccessibilitySupportEnabled')
 
diff --git a/lib/renderer/api/web-frame.ts b/lib/renderer/api/web-frame.ts
index ca640c30a..4d3b06489 100644
--- a/lib/renderer/api/web-frame.ts
+++ b/lib/renderer/api/web-frame.ts
@@ -85,15 +85,6 @@ function getWebFrame (context: Window) {
   return context ? new WebFrame(context) : null
 }
 
-const promisifiedMethods = new Set<string>([
-  'executeJavaScript',
-  'executeJavaScriptInIsolatedWorld'
-])
-
-for (const method of promisifiedMethods) {
-  (WebFrame as any).prototype[method] = deprecate.promisify((WebFrame as any).prototype[method])
-}
-
 const _webFrame = new WebFrame(window)
 
 export default _webFrame
diff --git a/lib/renderer/web-view/web-view-impl.ts b/lib/renderer/web-view/web-view-impl.ts
index 6849f3061..34545167b 100644
--- a/lib/renderer/web-view/web-view-impl.ts
+++ b/lib/renderer/web-view/web-view-impl.ts
@@ -273,7 +273,7 @@ export const setupMethods = (WebViewElement: typeof ElectronInternal.WebViewElem
   }
 
   for (const method of asyncPromiseMethods) {
-    (WebViewElement.prototype as Record<string, any>)[method] = deprecate.promisify(createPromiseHandler(method))
+    (WebViewElement.prototype as Record<string, any>)[method] = createPromiseHandler(method)
   }
 }
```

**Comment by miniak** [comment]
@codebytere shouldn't we also remove the documentation for the callback versions?

**Comment by codebytere** [comment]
I need to fix the smoke tests here 👇

**Comment by miniak** [comment]
> I need to fix the smoke tests here 👇
yeah, when I tried locally what happens when the docs are updated, I could see a lot of those blowing up :)

**Comment by codebytere** [comment]
Fixed now pending a issue with `electron-ts-defs`; we need to update such that Promise inner types are correctly parsed as arrays instead of raw types unconditionally.

cc @MarshallOfSound

File: docs/api/session.md [hunk 1]
```diff
@@ -153,85 +118,7 @@ Returns `Promise<void>` - resolves when the storage data has been cleared.
...
-   Examples:
-     "192.168.1.1/16", "fefe:13::abc/33".
-
-* `<local>`
-
-   Match local addresses. The meaning of `<local>` is whether the
-   host matches one of: "127.0.0.1", "::1", "localhost".
-
-**[Deprecated Soon](modernization/promisification.md)**
+Writes any unwritten DOMStorage data to disk.s
```
Comment (miniak) [inline docs/api/session.md]: typo disk.s

**Pull request merged**